crates/bevy_core_pipeline/src/tonemapping/mod.rs

```diff
@@ -0,0 +1,145 @@
+mod node;
+
+use bevy_ecs::query::QueryItem;
+use bevy_render::camera::Camera;
+use bevy_render::extract_component::{ExtractComponent, ExtractComponentPlugin};
+pub use node::TonemappingNode;
+
+use bevy_app::prelude::*;
+use bevy_asset::{load_internal_asset, HandleUntyped};
+use bevy_ecs::prelude::*;
+use bevy_render::renderer::RenderDevice;
+use bevy_render::texture::BevyDefault;
+use bevy_render::{render_resource::*, RenderApp};
+
+use bevy_reflect::TypeUuid;
+
+use crate::fullscreen_vertex_shader::fullscreen_shader_vertex_state;
+
+const TONEMAPPING_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 17015368199668024512);
+
+const TONEMAPPING_SHARED_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 2499430578245347910);
+
+const BLIT_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 2982361071241723543);
+
+pub struct TonemappingPlugin;
+
+impl Plugin for TonemappingPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(
+            app,
+            TONEMAPPING_SHADER_HANDLE,
+            "tonemapping.wgsl",
+            Shader::from_wgsl
+        );
+        load_internal_asset!(
+            app,
+            TONEMAPPING_SHARED_SHADER_HANDLE,
+            "tonemapping_shared.wgsl",
+            Shader::from_wgsl
+        );
+        load_internal_asset!(app, BLIT_SHADER_HANDLE, "blit.wgsl", Shader::from_wgsl);
+
+        app.add_plugin(ExtractComponentPlugin::<Tonemapping>::default());
+
+        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app.init_resource::<TonemappingPipeline>();
+        }
+    }
+}
+
+#[derive(Resource)]
+pub struct TonemappingPipeline {
+    hdr_texture_bind_group: BindGroupLayout,
+    tonemapping_pipeline_id: CachedRenderPipelineId,
+    blit_pipeline_id: CachedRenderPipelineId,
+}
+
+impl FromWorld for TonemappingPipeline {
+    fn from_world(render_world: &mut World) -> Self {
+        let tonemap_texture_bind_group = render_world
```

Comment:
This is a bind group layout, not a bind group, so the variable name is misleading.